pkg/file/pipeline/pipeline.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package pipeline
+
+import (
+	"github.com/ethersphere/bee/pkg/storage"
+)
+
+type pipeWriteArgs struct {
+	ref  []byte
+	span []byte
+	data []byte //data includes the span too
+}
+
+type Pipeline struct {
+	head Interface
+}
+
+func (p *Pipeline) Write(b []byte) (int, error) {
+	return p.head.Write(b)
+}
+
+func (p *Pipeline) Sum() ([]byte, error) {
+	return p.head.Sum()
+}
```

Comment:
Pipeline type just wraps two methods of the Interface in a concrete type. I am not sure why it is needed, and it can be completely removed.